crates/volta-core/src/tool/package_global/metadata.rs

```diff
@@ -91,6 +115,29 @@ impl BinConfig {
         serde_json::from_reader(config).with_context(|| ErrorKind::ParseBinConfigError)
     }
 
+    pub fn from_file_if_exists<P>(file: P) -> Fallible<Option<Self>>
+    where
+        P: AsRef<Path>,
+    {
+        match File::open(&file) {
+            Err(error) => {
+                if error.kind() == io::ErrorKind::NotFound {
+                    Ok(None)
+                } else {
+                    Err(VoltaError::from_source(
+                        error,
+                        ErrorKind::ReadBinConfigError {
+                            file: file.as_ref().to_owned(),
+                        },
+                    ))
+                }
+            }
+            Ok(config) => serde_json::from_reader(config)
+                .with_context(|| ErrorKind::ParseBinConfigError)
+                .map(|config| Some(config)),
```

Comment:
Same suggestion here about the unnecessary closure:

```suggestion
                .map(Some),
```